extensions/wikia/Search/classes/IndexService/AllUnifiedSearch.php

```diff
@@ -49,7 +49,30 @@ public function execute() {
 			}
 		}
 
+		$result['is_disabled_in_global_search'] = $this->isDiabledInGlobalSearch();
 		return $result;
 	}
 
+	/**
+	 * Calculate flag for deciding to include page in global search.
+	 *
+	 * @return array
+	 */
+	protected function isDiabledInGlobalSearch() {
+		global $wgExcludeWikiFromSearch;
+		global $wgForceWikiIncludeInSearch;
+
+		if ( !empty( $wgExcludeWikiFromSearch ) ) {
+			return true;
+		}
+		if ( !empty( $wgForceWikiIncludeInSearch ) ) {
+			return false;
+		}
+
+		$service = $this->getService();
+		$wikiData = $service->getApiStatsForWiki();
+		$wikiArticles = $wikiData['query']['statistics']['articles'];
```

Comment:
kinda overkill, `SiteStats::articles` does the same

Reply:
Thanks :)